File: projects/01/rectangle_area.rb
#! /usr/bin/ruby
# rectangle_area.rb computes the area of a rectangle given its width and height
# 
# Input: the width and height of a rectangle
# Precondition: the width and height are not negative
# Output: the area of the rectangle.
#
# Author: ZeAi Sun
# Date: Jan 12, 2023
###############################################################

# function rectangleArea returns a rectangle's area, given its width and height
# Parameters: w and h, both numbers
# Precondition: w > 0, h > 0
# Returns: the area of a rectangle whose width is w and height is h.
def rectangleArea(w, h)  # define the function
    w * h
end

if __FILE__ == $0  # the main program
   puts "To compute the area of a rectangle,"  # display the message
   print " enter its width: "
   width = gets.chomp.to_f  # get the width input from the user
   print " enter its height: "
   height = gets.chomp.to_f  # get the height input from the user
   print "The circle's area is: "
   puts rectangleArea(width, height)  # call the function and display the result
end

File: projects/01/RectangleArea.java
/* RectangleArea.java computes the area of a rectangle.
 *
 * Input: The width and height of the rectangle.
 * Precondition: The width and height are positive numbers.
 * Output: The area of the rectangle.
 *
 * Author: ZeAi Sun
 * Date: Jan 12, 2023
 **********************************************************/

import java.util.Scanner;  // include class for Scanner

public class RectangleArea {

     /* function RectangleArea() computes a rectangle's area, given its width and height.
      * Parameter: w and h, both doubles
      * Precondition: w and h are not negative.
      * Returns: the area of the rectangle
      */
     public static double RectangleArea(double w, double h) {
        return w * h;               // return an expression
     } // RectangleArea method
	
  // main program
  public static void main(String[] args) {
      // prompt for width
      System.out.println("\n\nTo compute the area of a rectangle,");
      System.out.print(" enter its width: ");

      // Create a connection named keyboard to standard in
      Scanner keyboard = new Scanner(System.in);

      //Get the number from the user
      double width = keyboard.nextDouble();

      // prompt for height and get the number from the user
      System.out.print(" enter its height: ");
      double height = keyboard.nextDouble();

      // output area
      System.out.println("\nThe area is " + RectangleArea(width, height) + "\n");
      System.out.printf("The area is %f\n\n", RectangleArea(width, height));
      System.out.printf("The area is %.15f\n\n", RectangleArea(width, height));
  } // main method

} // class RectangleArea
